### PR TITLE
Add documentation for ReflectionProperty::getDefaultValue/hasDefaultValue

### PR DESCRIPTION
**Pull request by beberlei**
Its a bit early for this, please don't merge yet, but this related to https://github.com/php/php-src/commit/7a52bf21d358567980bc20a95cef730bc9e9f00d

### CODE REVIEW

**Comment by nikic** [comment]
Travis failure legitimate.

File: reference/reflection/reflectionproperty/getdefaultvalue.xml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>mixed</type><methodname>ReflectionProperty::getDefaultValue</methodname>
+   <void />
+  </methodsynopsis>
+  <para>
+	  Gets the implicit or explicitly declared default value for a property.
```
Comment (nikic) [inline reference/reflection/reflectionproperty/getdefaultvalue.xml]: Looks like some tabs snuck into the indentation.

File: reference/reflection/reflectionproperty/getdefaultvalue.xml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  &reftitle.parameters;
+  &no.function.parameters;
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+	  The default value if the property has any default value (including &null;).
+	  If there is no default value, then &null; is returned. You cannot differentiate
+	  between a valid &null; default value and a non-nullable typed property.
```
Comment (nikic) [inline reference/reflection/reflectionproperty/getdefaultvalue.xml]: I'd write "between a null default value and an uninitialized typed property" here. Whether the property is nullable or not doesn't matter, both can either have or not have a default value.

File: reference/reflection/reflectionproperty/hasdefaultvalue.xml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>bool</type><methodname>ReflectionProperty::hasDefaultValue</methodname>
+   <void />
+  </methodsynopsis>
+  <para>
+	  Checks whether the property was declared with a default value. &null; is a valid default value
+	  that can be implicitly set for properties that are nullable. Only typed properties that are
+	  not nullable may have no default value.
```
Comment (nikic) [inline reference/reflection/reflectionproperty/hasdefaultvalue.xml]: Again this is a bit misleading. None of this documentation should contain the word "nullable" really. Something like...

> Checks whether the property was declared with a default value, including an implicit &null; default value. Only returns false for typed properties without default value (or dynamic properties).

File: reference/reflection/reflectionproperty/hasdefaultvalue.xml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+var_dump($ro->getProperty('baz')->hasDefaultValue());
+var_dump($ro->getProperty('boing')->hasDefaultValue());
+?>
+]]>
+	</programlisting>
+	&example.outputs;
+	<screen>
+<![CDATA[
+bool(true)
+bool(true)
```
Comment (nikic) [inline reference/reflection/reflectionproperty/hasdefaultvalue.xml]: This doesn't return true, it returns `false`.

File: reference/reflection/reflectionproperty/getdefaultvalue.xml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>mixed</type><methodname>ReflectionProperty::getDefaultValue</methodname>
+   <void />
+  </methodsynopsis>
+  <para>
+   Gets the implicit or explicitly declared default value for a property.
```
Comment (cmb69) [inline reference/reflection/reflectionproperty/getdefaultvalue.xml]: implicitly?

File: reference/reflection/reflectionproperty/getdefaultvalue.xml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+ <refsect1 role="parameters">
+  &reftitle.parameters;
+  &no.function.parameters;
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   The default value if the property has any default value (including &null;).
+   If there is no default value, then &null; is returned. You cannot differentiate
```
Comment (cmb69) [inline reference/reflection/reflectionproperty/getdefaultvalue.xml]: No "you" in man pages, please.
```suggestion
   If there is no default value, then &null; is returned. It is not possible to differentiate
```

File: reference/reflection/reflectionproperty/getdefaultvalue.xml [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+   between a &null; default value and an unitialized typed property.
+   Use <methodname>ReflectionClass::hasDefaultValue</methodname> to detect the difference.
+  </para>
+ </refsect1>
+
+ <refsect1 role="examples">
+  &reftitle.examples;
+  <para>
+   <example>
+ <title><methodname>ReflectionClass::getDefaultValue</methodname> example</title>
```
Comment (cmb69) [inline reference/reflection/reflectionproperty/getdefaultvalue.xml]: Nit: indentation

File: reference/reflection/reflectionproperty/getdefaultvalue.xml [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+ <programlisting role="php">
+<![CDATA[
+<?php
+class Foo {
+ public $bar = 1;
```
Comment (cmb69) [inline reference/reflection/reflectionproperty/getdefaultvalue.xml]: nit: indentation (I'd use four space in code)

File: reference/reflection/reflectionproperty/getdefaultvalue.xml [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+ </screen>
+   </example>
+  </para>
+ </refsect1>
+
+ <refsect1 role="seealso">
+  &reftitle.seealso;
+  <para>
+   <simplelist>
+ <member><methodname>ReflectionProperty::hasDefaultValue</methodname></member>
```
Comment (cmb69) [inline reference/reflection/reflectionproperty/getdefaultvalue.xml]: nit: identation

File: reference/reflection/reflectionproperty/hasdefaultvalue.xml [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>bool</type><methodname>ReflectionProperty::hasDefaultValue</methodname>
+   <void />
+  </methodsynopsis>
+  <para>
+   Checks whether the property was declared with a default value, including an implicit
+   &null; default value. Only returns false for typed properties without default value (or
```
Comment (cmb69) [inline reference/reflection/reflectionproperty/hasdefaultvalue.xml]:
```suggestion
   &null; default value. Only returns &false; for typed properties without default value (or
```

File: reference/reflection/reflectionproperty/hasdefaultvalue.xml [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+ <refsect1 role="parameters">
+  &reftitle.parameters;
+  &no.function.parameters;
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   &true; if the property has any default value (including &null;) it returns true,
+   if the property is typed without a default value declared or a dynamic property returns false
```
Comment (cmb69) [inline reference/reflection/reflectionproperty/hasdefaultvalue.xml]: This sentence looks not quite right.

File: reference/reflection/reflectionproperty/getdefaultvalue.xml (hunk 5 above)
Comment (cmb69) [reply reference/reflection/reflectionproperty/getdefaultvalue.xml:17]: nope :)

**Review by cmb69 — APPROVED** [review]

**Comment by cmb69** [comment]
Applied as http://svn.php.net/viewvc?view=revision&revision=351090.

**Pull request closed without merging**